back-end/hub-core/src/main/java/io/apicurio/hub/core/integrations/http/HttpEventSink.java

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apicurio.hub.core.integrations.http;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.apicurio.hub.core.integrations.ApicurioEventType;
+import io.apicurio.hub.core.integrations.EventSink;
+
+@ApplicationScoped
+public class HttpEventSink implements EventSink {
+
+    private static final Logger log = LoggerFactory.getLogger(HttpEventSink.class);
+
+    private HttpClient httpClient;
+
+    @Inject
+    HttpSinksConfiguration configuration;
+
+    @Override
+    public String name() {
+        return "http sink";
+    }
+
+    @Override
+    public boolean isConfigured() {
+        return configuration.isConfigured();
+    }
+
+    @Override
+    public void handle(ApicurioEventType type, String designId, byte[] data) {
+
+        log.info("Firing event " + type);
+
+        for (HttpSinkConfiguration httpSink : configuration.httpSinks()) {
+            sendEventHttp(type, httpSink, data);
+        }
+
+    }
+
+    private void sendEventHttp(ApicurioEventType type, HttpSinkConfiguration httpSink, byte[] data) {
+        try {
+            log.debug("Sending event to sink "+httpSink.getName());
+
+            HttpPost request = new HttpPost(httpSink.getEndpoint());
+            request.addHeader("ce-id", UUID.randomUUID().toString());
+            request.addHeader("ce-specversion", "1.0");
+            request.addHeader("ce-source", "apicurio-registry");
+            request.addHeader("ce-type", type.cloudEventType());
+            request.addHeader("content-type", ContentType.APPLICATION_JSON.getMimeType());
+            request.setEntity(new ByteArrayEntity(data, ContentType.APPLICATION_JSON));
+            try {
+                HttpResponse response = getHttpClient().execute(request);
+                if (response.getStatusLine().getStatusCode() < 200 || response.getStatusLine().getStatusCode() > 299) {
+                    log.warn("Http sink {} return status code {}", httpSink.getName(), response.getStatusLine().getStatusCode());
+                }
+            } catch (IOException e) {
+                log.error("Error sending event to " + httpSink.getEndpoint(), e);
+            }
+
+        } catch (Exception e) {
+            log.error("Error sending http event", e);
+        }
+    }
+
+    private synchronized HttpClient getHttpClient() {
+
+        if (httpClient == null) {
+            httpClient = HttpClientBuilder.create().build();
+        }
+        return httpClient;
+    }
```

Comment:
Initialize in a `@PostConstruct` method to avoid the synchronized method?

Reply:
I think that would result in the http client always being initialized, even when not configuring it. Which is what I tried to avoid

Reply:
You could create the client only if `configuration.isConfigured()` returns `true`, no?

Reply:
```java
@PostConstruct
void init() {
    if (configuration.isConfigured()) {
        httpClient = HttpClientBuilder.create().build();
    }
}
```